docs/usage.md

```diff
@@ -0,0 +1,236 @@
+
+- [k6bench.gnuplot](/k6bench.gnuplot): this is a [gnuplot](http://www.gnuplot.info/) script that generates a graph from the CSV results created by `k6bench.sh`.
+  It doesn't create the prettiest graphs, but it was chosen because it's scriptable, so it can be easily automated.
+
+
+## Dependencies
+
+The benchmarks depend on a few utilities to collect the performance results:
+
+- [`smem`](https://www.selenic.com/smem/) for RAM usage, specifically proportional set size (PSS).
+
+  As it turns out, it's rather tricky to determine the precise physical RAM usage of a single process in Linux. Tools like `ps` and `htop` report the resident set size (RSS), which includes the size of any shared libraries the process may be using, and is not an accurate value of how much physical memory the process is using. The PSS value is more precise, but is more difficult to measure. `smem` is seemingly the only tool that simplifies tracking this, but it's unfortunately written in Python, so its own performance is not great, and it can use a lot of CPU and take some time to generate the result.
+
+- [`jq`](https://stedolan.github.io/jq/) is used to extract the VUs and RPS values from the JSON output returned by the k6 REST API.
+
+- [`curl`](https://curl.se/) to make requests to the k6 REST API.
+
+- GNU `coreutils` like `ps`, `top`, `cat`, `awk`, and Bash itself, of course.
+
+
+## Benchmarking strategy
+
+We will run one test at a time, to avoid overloading the SUT and skewing the results.
+
+We won't run all scripts on all machines, but will use the following strategy:
+
+- From the machine with the lowest specs, i.e. `m5.large`, we will only run the `website.js` script, with a high sleep time between iterations in order to not overwhelm the load generator. The goal of this test is to track overall CPU and memory usage, not reach maximum throughput.
+
+- From the middle-spec'd machine, i.e. `m5.4xlarge`, we will run the `website.js` script with more VUs and a lower sleep time between iterations. This way we can observe the change in resource consumption and throughput by using the same script with an increased load.
+
+- From the machine with the highest specs, i.e. `m5.24xlarge`, we will run the `website.js` script with even more VUs and a very low sleep time between iterations. We will also run the RPS optimized and file upload test scripts, so we can observe the performance of k6 when it's allowed to fully utilize the system resources.
+
+
+## Running the benchmarks
+
+Assuming we have the testing environment ready, we can now proceed to run the benchmarks themselves.
+
+1. Establish an SSH connection to the load generator machine you wish to use first.
+
+2. Install k6 and all dependencies for running the benchmarks. Ideally these would be pre-installed in the load generator system image by Ansible, but if not, let's install them manually:
```

Comment:
>by Ansible

Do we have it somewhere? I don't see the provisioning script in this repo.

Reply:
No, not at all. The loadgen machines were provisioned manually, which is what this explains. I just wanted to mention that once we automate this, these dependencies should be pre-installed in the image by whatever process provisions it, which would likely be Ansible.